dask/array/reshape.py

```diff
@@ -226,6 +231,36 @@ def reshape(x, shape, merge_chunks=True):
         x = x.rechunk({i: 1 for i in range(din - dout)})
 
     inchunks, outchunks = reshape_rechunk(x.shape, shape, x.chunks)
+    # Check output chunks are not too large
+    max_chunksize_in_bytes = np.prod([max(i) for i in outchunks]) * x.dtype.itemsize
+    if (
+        limit is None
+        and config.config["array"]["slicing"]["split-large-chunks"] is not False
+    ):
+        limit = parse_bytes(config.get("array.chunk-size"))
+    if max_chunksize_in_bytes > limit:
+        # Leave chunk sizes unaltered where possible
+        matching_chunks = Counter(inchunks) & Counter(outchunks)
+        chunk_plan = []
+        for out in outchunks:
+            if out in matching_chunks:
+                chunk_plan.append(out)
+                # Remove match from list
+                if matching_chunks[out] <= 1:
```

Comment:
```suggestion
                if matching_chunks[out] == 1:
```

This counter should never reach 0, correct?

Reply:
Ok, true. It should get down to 1 and then that item is removed entirely.